packages/db/src/core/cli/commands/link/index.ts

```diff
@@ -1,36 +1,76 @@
+import { slug } from 'github-slugger';
 import prompts from 'prompts';
 import type { Arguments } from 'yargs-parser';
 import { PROJECT_ID_FILE, getSessionIdFromFile } from '../../../tokens.js';
 import { getAstroStudioUrl } from '../../../utils.js';
 import { MISSING_SESSION_ID_ERROR } from '../../../errors.js';
+import { basename } from 'node:path';
+import ora from 'ora';
 
 export async function cmd({ flags }: { config: AstroConfig; flags: Arguments }) {
-	const linkUrl = new URL(getAstroStudioUrl() + '/auth/cli/link');
 	const sessionToken = await getSessionIdFromFile();
 	if (!sessionToken) {
 		console.error(MISSING_SESSION_ID_ERROR);
 		process.exit(1);
 	}
-	let body = { id: flags._[4] } as {
-		id?: string;
-		projectIdName?: string;
-		workspaceIdName?: string;
-	};
-	if (!body.id) {
-		const workspaceIdName = await promptWorkspaceName();
-		const projectIdName = await promptProjectName();
-		body = { projectIdName, workspaceIdName };
+	await promptBegin();
+	const isLinkExisting = await promptLinkExisting();
+	if (isLinkExisting) {
+		const projectId = await promptExistingProjectName();
+		return await linkProject(projectId);
+	}
+
+	const isLinkNew = await promptLinkNew();
+	if (isLinkNew) {
+		const newProjectName = await promptNewProjectName();
+		const newProjectRegion = await promptNewProjectRegion();
+		const newProjectId = await createNewProject({name: newProjectName, region: newProjectRegion});
+		// TODO(fks): Actually listen for project creation before continuing
+		// This is just a dumb spinner that roughly matches database creation time.
+		const spinner = ora('Creating new project...');
+		await new Promise((r) => setTimeout(r, 4000));
+		spinner.succeed('Project created!');
+		return await linkProject(newProjectId);
+	}
+}
+
+async function linkProject(id: string) {
+	await mkdir(new URL('.', PROJECT_ID_FILE), { recursive: true });
+	await writeFile(PROJECT_ID_FILE, `${id}`);
+	console.info('Project linked.');
+}
+
+export async function createNewProject({name, region}: {name: string, region: string}): Promise<string> {
+	const linkUrl = new URL(getAstroStudioUrl() + '/api/cli/projects.create');
+	const response = await fetch(linkUrl, {
+		method: 'POST',
+		headers: {
+			Authorization: `Bearer ${await getSessionIdFromFile()}`,
+			'Content-Type': 'application/json',
+		},
+		body: JSON.stringify({ name, region }),
+	});
+	if (!response.ok) {
+		console.error(`Failed to create project: ${response.status} ${response.statusText}`);
+		process.exit(1);
 	}
+	const { data } = await response.json();
+	return data.id;
+}
+
+export async function promptExistingProjectName(): Promise<string> {
+	const linkUrl = new URL(getAstroStudioUrl() + '/api/cli/projects.list');
 	const response = await fetch(linkUrl, {
 		method: 'POST',
 		headers: {
 			Authorization: `Bearer ${await getSessionIdFromFile()}`,
 			'Content-Type': 'application/json',
 		},
-		body: JSON.stringify(body),
+		body: JSON.stringify({}),
```

Comment:
nit: should this be a GET? Seems odd to post an empty body to get a list of projects

Reply:
Because we're already maintaining TRPC, we follow an RPC-like approach to this API as well, which effectively says that request method doesn't matter (so POST is okay even if you're just getting data).